pandas/core/arrays/timedeltas.py

```diff
@@ -715,15 +720,14 @@ def __rfloordiv__(self, other):
         if is_scalar(other):
             if isinstance(other, self._recognized_scalars):
                 other = Timedelta(other)
-                if other is NaT:
+                if cast("Timedelta | NaTType", other) is NaT:
```

Comment:
this is made necessary bc the annotation on `Timedelta.__new__` is not correctly understood (has a type:ignore) and is waiting for a fix upstream. can you add a comment to that effect

Reply:
Yes, will add comments